src/internal/connector/exchange/api/events.go

```diff
@@ -182,23 +178,54 @@ func (c Events) GetAddedAndRemovedItemIDs(
 		return nil, nil, DeltaUpdate{}, err
 	}
 
-	var errs *multierror.Error
+	var (
+		resetDelta bool
+		errs       *multierror.Error
+	)
 
-	options, err := optionsForEventsByCalendar([]string{"id"})
+	options, err := optionsForEventsByCalendarDelta([]string{"id"})
 	if err != nil {
 		return nil, nil, DeltaUpdate{}, err
 	}
 
-	builder := service.Client().UsersById(user).CalendarsById(calendarID).Events()
+	if len(oldDelta) > 0 {
+		builder := users.NewItemCalendarsItemEventsDeltaRequestBuilder(oldDelta, service.Adapter())
+		pgr := &eventPager{service, builder, options}
+
+		added, removed, deltaURL, err := getItemsAddedAndRemovedFromContainer(ctx, pgr)
+		// note: happy path, not the error condition
+		if err == nil {
+			return added, removed, DeltaUpdate{deltaURL, false}, errs.ErrorOrNil()
+		}
+		// only return on error if it is NOT a delta issue.
+		// on bad deltas we retry the call with the regular builder
+		if graph.IsErrInvalidDelta(err) == nil {
+			return nil, nil, DeltaUpdate{}, err
+		}
+
+		resetDelta = true
+		errs = nil
+	}
+
+	// Graph SDK only supports delta queries against events on the beta version, so we're
+	// manufacturing use of the beta version url to make the call instead.
+	// See: https://learn.microsoft.com/ko-kr/graph/api/event-delta?view=graph-rest-beta&tabs=http
+	// Note that the delta item body is skeletal compared to the actual event struct.  Lucky
+	// for us, we only need the item ID.  As a result, even though we hacked the version, the
```

Comment:
can you check that items don't get skipped accidentally due to us selecting the bare minimum fields? Tbh, I'm not sure if this is a feature or a "bug" in how Graph works, but using `$select` can sometimes cause a delta endpoint to skip returning results if they don't match what's in the `$select`. For example, not having `isRead` in the select for emails will cause emails where the `isRead` flag has been changed to be skipped because it doesn't update the mod time. This will take some playing around and is probably easiest to do with GraphExplorer/postman

Reply:
Hmm, good call out.  I'll do some manual testing.  This might be especially interesting in this case, because the delta body for an event are only able to contain the id, startTime, and endTime.  Hopefully the delta will catch changes on unrepresented properties, like renames and etc.

Reply:
Alright, two findings.
1. everything works.  Changes to name, desc, attendees, etc, are all picked up by the delta.
2. we're not allowed to select values in the first place (see the error below).  If anything, we're lucky that the client is tossing the use of options for us.
```json
    "error": {
        "code": "ErrorInvalidUrlQuery",
        "message": "The following parameters are not supported with change tracking over the 'SyncEvents' resource: '$orderby, $filter, $select, $expand, $search'."
    }
```